lib/application/logic/user_state.dart

```diff
@@ -7,18 +7,90 @@ class UserState extends Equatable {
   final String companyUrl;
   final String phoneNumber;
   final bool userData;
+  final bool userUpdated;
   final dynamic error;
+  final bool loading;
+  final String currentPassword;
+  final bool passwordProvider;
```

Comment:
Ajde umjesto `passwordProvider` i `googleProvider` napravi nekakav enum koji će za sada imati dvije vrijednosti: `email` i `social`. U budućnosti ćemo vrlo vjerojatno morati ubaciti i apple sign in, pa će onda biti jednostavnije implementirati